packages/jira-adapter/src/workspace_id.ts

```
@@ -34,13 +35,21 @@ const WORKSPACE_RESPONSE_SCHEME = Joi.object({
 
 const isWorkspaceResponse = createSchemeGuard<WorkspaceResponse>(WORKSPACE_RESPONSE_SCHEME, 'Received invalid workspace response')
 
-export const getWorkspaceId = async (client: JiraClient): Promise<string | undefined> => {
-  const response = await client.getSinglePage({
-    url: '/rest/servicedeskapi/assets/workspace',
-  })
-  if (!isWorkspaceResponse(response.data)) {
-    log.trace('Received invalid workspace response %o', response.data)
+export const getWorkspaceId = async (client: JiraClient, config: JiraConfig): Promise<string | undefined> => {
+  try {
+    if (!config.fetch.enableJsmExperimental) {
+      return undefined
+    }
+    const response = await client.getSinglePage({
+      url: '/rest/servicedeskapi/assets/workspace',
+    })
+    if (!isWorkspaceResponse(response.data)) {
+      log.debug('Received invalid workspace response %o', response.data)
+      return undefined
+    }
+    return response.data.values[0].workspaceId
+  } catch (e) {
+    log.debug(`Failed to get workspace id: ${e}`)
     return undefined
```

Could you please explain the try..catch reasoning here?
In case we fail here, wouldn't we like the operation to fail? can we proceed as if nothing happened?
In case we "swallow" the error, we should at least log an error/warn

---

It doesn't suppose to happen - 
But customers that have JSM Free and don't have JSM premium will call this endpoint add get `404 not found` of `401 Unauthorized`, but I am afraid that it will throw an error for some unknown reason and I don't want their fetch\deploy to fail since they don't use the workspaceId anyway.